interfaces/builtin/unity8.go

```diff
@@ -40,6 +128,26 @@ func (iface *Unity8Interface) PermanentPlugSnippet(plug *interfaces.Plug, securi
 }
 
 func (iface *Unity8Interface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		oldTags := []byte("###SLOT_SECURITY_TAGS###")
+		newTags := slotAppLabelExpr(slot)
+		snippet := bytes.Replace([]byte(unity8ConnectedPlugAppArmor), oldTags, newTags, -1)
+
+		// FIXME: Until we decide whether unity8 is going to use Snappy-style
+		//        appIDs (snap.NAME.COMMAND) or Touch-style ones
+		//        (NAME_COMMAND_REVISION), we'll use a simpler *NAME* glob
+		//        here. This glob does not provide isolation because 'foo' can
+		//        access 'barfoobaz'. The base declaration restriction cannot
+		//        be lifted until this is properly mediated.
+		appidsOld := []byte("###PLUG_DBUS_APPIDS###")
+		appidsNew := fmt.Sprintf("*%s*", dbus.SafePath(plug.Snap.Name()))
```

Comment:
This is tricky. I have a strong preference that unity8 migrate away from the old Touch APP_IDs and use proper ids that are compatible with snappy and I therefore very much dislike this rule-- which is why there is all the scary language around it.

I think it is possible to do this with an attribute, and even thought about it before, but I was disinclined to add an attribute because I then felt this temporary code was potentially more permanent and wouldn't be able to be taken away easily. In other words, because of my strong preference for moving away from Touch IDs, I didn't want them to gradually go away-- I wanted them to go away ASAP and not linger in the snapd codebase and requiring the migration to happen before the base declaration is updated is incentive to do that.

I agreed to the current approach with the understanding that the broken rules would be incompatibly removed before the base declaration was updated for general use, and mentioned that any snaps may break when core changes and the unity8-session (slot implementation), snaps plugging unity8 and (I guess) the platform snap are updated (ie, your concerns). Since Canonical is driving the implementation of the interface, the unity8-session slot implementation and the platform snap, the agreement was that any snaps plugging this interface would also have to come from Canonical, and Canonical would be responsible for the transition and have to handle any breakage (there are <10 snaps currently that have a snap declaration to plugs the unity8 interface at this time). The Personal team agreed to these terms.